### ISSUE TITLE
value_counts() and groupby() treat the bytes type input values as 'missing'

### ISSUE THREAD

**Issue by And0k** [vaexio/vaex#729]
Bytes not supported? 
```python
ds = vaex.from_arrays(my_array=np.array(['1-2-3', '1-2-4', '1-2-5', '1-2-5', '1-2-6', '127', '127'], dtype=bytes))   # The following will find duplicates if remove "dtype=bytes"
uniques = ds['my_array'].value_counts()
>>>uniques
Out[1]:
missing    7
dtype: int64
```

Or
```python
>>>ds.groupby(by='my_array', agg={'my_array_cnt': vaex.agg.count('my_array')})
Out[1]: 
  #  my_array      my_array_cnt
  0  null                        7
```

---
vaex-1.0.0-beta.6-windows, python 3.7.6

**Comment by JovanVeljanoski** [comment]
Hi @And0k 

Indeed, bytes are not supported for now. Please convert the strings. 

There are provisional tests, but we have not gotten around to implementing them (e.g. #325).

**Comment by And0k** [comment]
Then it is strange that `ds['my_array'] = ds['my_array'].astype('O')` allows above mentioned operations and they return bytes (`uniques['my_array'].values.dtype.type`)

**Comment by JovanVeljanoski** [comment]
Well, dtype object are treated differently, and their usage is not recommended. We only support them for pandas compatibility. Maybe @maartenbreddels can shed some more light on this. 
If you convert the bytes to strings, you should get optimal performance (if that is allowed by your usecase).

**Comment by maartenbreddels** [comment]
Yes, we cannot directly support bytes now, that will be easier to do via arrow (once we merge that). However, casting to dtype=O is a nice workaround. Note however that that will have a performance penalty (the GIL cannot be released). So it will not work multithreaded.

Let's keep this open to inspect why it is reported as missing values, instead of giving an error.

Thanks for the reporting! Very much appreciated.
